### ISSUE TITLE
Current version on docs is always one version behind in automated release

### ISSUE THREAD

**Issue by mtscout6** [react-bootstrap/react-bootstrap#547]
![](http://cl.ly/image/0K0d2c2P1G0v/Image%202015-04-21%20at%208.12.38%20AM.png)

The version displayed at the footer of the online documentation is always one version behind when we release. 0.21.0 was just released this morning and it reads 0.20.3, and prior to the release it was 0.20.2.

I'm pretty sure the culprit is here: https://github.com/react-bootstrap/react-bootstrap/blob/master/docs/src/PageFooter.js#L2

I'm thinking that this module get's imported into the NodeJS Runtime prior to the version bump here: https://github.com/react-bootstrap/react-bootstrap/blob/master/tools/release-scripts/version-bump.js

The best way I can think of to change this is to have the docs generation get executed in a separate process after the version bump occurs. This change would be here: https://github.com/react-bootstrap/react-bootstrap/blob/master/tools/build.js#L32

Basically don't import the docs build script, instead execute `npm run docs-build` as specified in the [package.json](https://github.com/react-bootstrap/react-bootstrap/blob/v0.21.0/package.json#L15)

This should be a simple change should somebody wish to tackle their understanding of the release scripts.

**Comment by AlexKVal** [comment]
I'll try to tuckle with those maintainers tools.

Something like that. But I need to test it first. (and doing this not publish accidentally)
here [tools/build.js:32](https://github.com/react-bootstrap/react-bootstrap/blob/master/tools/build.js#L32)
```js
 let result = Promise.all([
  lib(),
  bower(),
  dist() 
  // removed docs()
])
.then(() => exec(`npm run docs-build`)) // <= added
.then(() => copy(distFolder, amdFolder));
```

**Comment by mtscout6** [comment]
`exec` returns a promise and the docs can run in parallel with everything else, I was thinking something like: 

```
 let result = Promise.all([
  lib(),
  bower(),
  dist(),
  exec(`npm run docs-build`)
])
.then(() => copy(distFolder, amdFolder));
```

**Comment by dozoisch** [comment]
@mtscout6 yeah that should do it.

**Comment by mtscout6** [comment]
@AlexKVal About testing your changes, I haven't figured out a good way to test these build/release scripts yet. May be a goal for another day.

**Comment by dozoisch** [comment]
I tests them by commenting out the push/publish (and the deletion of folders). This way you can check that everything else happened properly but yeah, needs automated way of doing it

**Comment by mtscout6** [comment]
What about adding a `--dry-run` flag, which when applied you skip any tag, push, commit, or publish operation?

**Comment by mtscout6** [comment]
Still not automated but still useful for testing this stuff.

**Comment by dozoisch** [comment]
Thatd be easy to do

**Comment by mtscout6** [comment]
I can tackle that today real quick. I just created an issue for it.

**Comment by mtscout6** [comment]
@AlexKVal I made a mistake. Replacing the docs build with `exec('npm run docs-build')` will result in a stack overflow of sorts. `npm run docs-build` invokes the file in which I said to include this script..... Basically you'd just keep spawning off process with no end in sight. I hope we didn't kill your machine causing you to be silent on this issue.

**Comment by dozoisch** [comment]
I don't see where the recursion is? 

`npm run docs-build` `tools/build.js` with `--doc-only` which doesn't not run `tools/release-scripts/release.js`

**Comment by dozoisch** [comment]
Maybe running `node run-babel docs/build.js` ?

Or actually it should be okay, since docs-build runs with docs-only and doesn't take the same branch in the file !

**Comment by AlexKVal** [comment]
I was too exhausted and had gone to bed that day and just before I'd been asleep, I thought exactly the same :)
> `exec` returns a promise, so I can
```js
bower(),
  dist(),
  exec(`npm run docs-build`) // <== place it here
])
.then(() => copy(distFolder, amdFolder));
```

And about this
> I hope we didn't kill your machine causing you to be silent on this issue.

LOL :smile: :smile: :smile: :+1: 

I'm sorry. I had a lot of work these days, so I've been able to get to this issue just now.

I like the idea to use `npm run docs-build` command because it is generalized and leads to decoupling.
I've been aware of recursion, and I thought... thought... and at the end... the fatigue had overcame me :)

Thank you for `--dry-run` option. It will help a lot of course.

**Comment by mtscout6** [comment]
Fixed by #578
